src/components/vm/vmReplaceSpiceDialog.jsx

```diff
@@ -0,0 +1,89 @@
+import { useDialogs } from 'dialogs.jsx';
+import { fmt_to_fragments } from 'utils.jsx';
+
+import { NeedsShutdownAlert } from '../common/needsShutdown.jsx';
+import { domainReplaceSpice } from '../../libvirtApi/domain.js';
+
+const _ = cockpit.gettext;
+
+export const ReplaceSpiceDialog = ({ vmName, vmId, connectionName, vmRunning }) => {
+    const Dialogs = useDialogs();
+    const [error, dialogErrorSet] = useState(null);
+    const [inProgress, setInProgress] = useState(false);
+
+    function onReplace() {
+        setInProgress(true);
+
+        return domainReplaceSpice({ connectionName, id: vmId })
+                .then(() => Dialogs.close())
+                .catch(exc => {
+                    // we don't know of any case where this would fail, so this is all belt-and-suspenders
+                    // still, give the user some hint how to go on
+                    console.warn("Failed to replace SPICE devices:", exc);
+                    dialogErrorSet({
+                        dialogError: _("Failed to replace SPICE devices"),
+                        dialogErrorDetail: fmt_to_fragments(
+                            _("Please see $0 how to reconfigure your VM manually."),
+                            <a href="https://access.redhat.com/solutions/6955095" target="_blank" rel="noopener noreferrer">
+                                https://access.redhat.com/solutions/6955095
+                            </a>),
+                    });
+                })
+                .finally(() => setInProgress(false));
+    }
+
+    return (
+        <Modal position="top" variant="small" isOpen onClose={Dialogs.close}
+           title={cockpit.format(_("Replace SPICE devices in VM $0"), vmName)}
+           footer={
+               <>
+                   <Button variant='primary'
+                           id="replace-spice-dialog-confirm"
+                           isDisabled={inProgress}
+                           onClick={onReplace}>
+                       {_("Replace")}
+                   </Button>
+                   <Button variant='link' onClick={Dialogs.close}>
+                       {_("Cancel")}
+                   </Button>
+               </>
+           }>
+            { vmRunning && !error && <NeedsShutdownAlert idPrefix="spice-modal" /> }
+            {error && <ModalError dialogError={error.dialogError} dialogErrorDetail={error.dialogErrorDetail} />}
+            <TextContent>
+                <Text>{_("Reconfigure the virtual machine for a host which does not support SPICE, for host upgrades or live migration:")}</Text>
+                <TextList>
+                    <TextListItem>{_("Convert SPICE graphics console to VNC.")}</TextListItem>
+                    <TextListItem>{_("Convert QXL video card to VGA.")}</TextListItem>
+                    <TextListItem>{_("Remove SPICE audio and host devices.")}</TextListItem>
+                </TextList>
+            </TextContent>
+        </Modal>
```

Comment:
These 11 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1387-20240122-073835-c9864bc3-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)